metadata.yaml

```diff
@@ -1,38 +1,27 @@
-# Copyright 2021 Canonical Ltd.
+# Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 name: mlmd
 summary: Record and retrieve metadata associated with ML workflows
-description: https://www.tensorflow.org/tfx/guide/mlmd
-series: [kubernetes]
-min-juju-version: "2.9.0"
+description: |
+  The ML Metadata operator provides means to store runtime information of a pipeline run in a
+  Metadata store. This information includes status of a task, availability of artifacts, custom properties, etc.
+  More info in https://github.com/google/ml-metadata/blob/master/g3doc/get_started.md
+containers:
+  mlmd-grpc-server:
+    resource: oci-image
+    mounts:
+      - storage: mlmd-data
+        location: /data
 resources:
   oci-image:
     type: oci-image
     description: Backing OCI image
     auto-fetch: true
     upstream-source: gcr.io/tfx-oss-public/ml_metadata_store_server:1.14.0
 provides:
-  grpc:
-    interface: grpc
-    schema:
-      v1:
-        provides:
-          type: object
-          properties:
-            service:
-              type: string
-            port:
-              type: string
-          required:
-          - service
-          - port
-    versions: [v1]
-    __schema_source: https://raw.githubusercontent.com/canonical/operator-schemas/master/grpc.yaml
-requires:
-  mysql:
-    interface: mysql
+  k8s-svc-info:
+    interface: k8s-service
```

Comment:
iiuc, `kfp-metadata-writer` and `envoy` need to `require` this interface now and use the library, since they have a relation to this charm with `grpc` interface that we are replacing here. Are we tracking this effort somewhere?